src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopyColumnMappingCollection.cs

```diff
@@ -189,6 +166,49 @@ internal void ValidateCollection()
                 }
             }
         }
+
+        void ICollection.CopyTo(Array array, int index) => NonGenericList.CopyTo(array, index);
+
+        int IList.Add(object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            return NonGenericList.Add(value);
+        }
+
+        bool IList.Contains(object value) => NonGenericList.Contains(value);
+
+        int IList.IndexOf(object value) => NonGenericList.IndexOf(value);
+
+        void IList.Insert(int index, object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            NonGenericList.Insert(index, value);
+        }
+
+        void IList.Remove(object value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            bool removed = _list.Remove(value);
+
+            // This throws on full framework, so it will also throw here.
+            if (!removed)
+            {
+                throw new ArgumentException(SR.Arg_RemoveArgNotFound);
+            }
+        }
+
+        private IList NonGenericList => _list;
```

Comment:
Why is the `NonGenericList` needed? 
In the code the `_list` is being referenced using `NonGenericList` and `_list` member. Can we have this being done consistently?